netadapter/server/grpcserver/protowire/message_transaction.go

```diff
@@ -93,7 +93,10 @@ func (x *TransactionMessage) fromWireMessage(msgTx *wire.MsgTx) {
 		LockTime:     msgTx.LockTime,
 		SubnetworkID: wireSubnetworkIDToProto(&msgTx.SubnetworkID),
 		Gas:          msgTx.Gas,
-		PayloadHash:  wireHashToProto(msgTx.PayloadHash),
 		Payload:      msgTx.Payload,
 	}
+
+	if msgTx.PayloadHash != nil {
```

Comment:
To be more consistent with the current design, can you prepare the payload hash (nil or not) above `*x = TransactionMessage{`?